src/lib/transition_frontier/snark_pool_refcount.ml

```diff
@@ -16,19 +19,69 @@ module Make (Inputs : Inputs_intf) :
   Transition_frontier_extension_intf0
   with type transition_frontier_breadcrumb := Inputs.Breadcrumb.t
    and type input = unit
-   and type view = unit = struct
+   and type view = int * int Inputs.Transaction_snark_work.Statement.Table.t =
+struct
   module Work = Inputs.Transaction_snark_work.Statement
 
-  type t = {ref_table: int Work.Table.t}
+  type t = int Work.Table.t
+
+  type view = int * int Work.Table.t
 
   type input = unit
 
-  type view = unit
+  let get_work (breadcrumb : Inputs.Breadcrumb.t) : Work.t Sequence.t =
+    let ledger = Inputs.Breadcrumb.staged_ledger breadcrumb in
+    let scan_state = Inputs.Staged_ledger.scan_state ledger in
+    let work_to_do =
+      Inputs.Staged_ledger.Scan_state.all_work_to_do scan_state
+    in
+    Or_error.ok_exn work_to_do
+
+  (** Returns true if this update changed which elements are in the table
+  (but not if the same elements exist with a different reference count) *)
+  let add_breadcrumb_to_ref_table table breadcrumb : bool =
+    Sequence.fold ~init:false (get_work breadcrumb) ~f:(fun acc work ->
+        match Work.Table.find table work with
+        | Some count ->
+            Work.Table.set table ~key:work ~data:(count + 1) ;
+            acc || false
+        | None ->
+            Work.Table.set table ~key:work ~data:1 ;
+            acc || true )
+
+  (** Returns true if this update changed which elements are in the table
+  (but not if the same elements exist with a different reference count) *)
+  let remove_breadcrumb_from_ref_table table breadcrumb : bool =
```

Comment:
This could return the number of removed work rather than whether the breadcrumb was removed, if we want a more granular measurement for when we want to collect. Probably doesn't matter that much though